src/efftester.ml

```diff
@@ -148,7 +167,7 @@ let type_to_str ?(effannot = false) typ =
 
 let eff_to_str ((ef, ev) : eff) = Printf.sprintf "(%B,%B)" ef ev
 
-(* BNF grammar:
+(* FIXME: BNF grammar:
```

Comment:
To fix this grammar I would add `constr (exp, exp, ...)` at the `exp` level, and also `begin match exp with pat -> arg | pat -> arg | ... end`.

In term of parenthesis placement, this would suggest that the printing of `Constructor` below does not need outer parentheses. (I used `begin .. end` around match above as this is closer to the Coq syntax that I like (in OCaml `begin .. end` and `( .. )` are equivalent anyway), but the parentheses that you used in your implementation are also fine.)